src/app/module/containers/module-content-page/module-content-page.component.ts

```diff
@@ -0,0 +1,55 @@
+import { Component, OnDestroy, OnInit } from '@angular/core';
+import { tap } from 'rxjs/operators';
+import { noop } from 'rxjs';
+import { ModuleContentService } from '../../services/module-content.service';
+import { Lesson, ModuleProgress } from '../../models/module';
+import { ModuleService } from '../../services/module.service';
+import { ActivatedRoute } from '@angular/router';
+
+@Component({
+  selector: 'module-content-page',
+  templateUrl: './module-content-page.component.html',
+  styleUrls: ['./module-content-page.component.scss']
+})
+export class ModuleContentPageComponent implements OnInit, OnDestroy {
+
+  lesson: Lesson;
+  moduleProgress: ModuleProgress;
+  lessonDone: boolean;
+
+  subscriptions = [];
+
+  constructor(
+    private moduleContentService: ModuleContentService,
+    private moduleService: ModuleService,
+    private route: ActivatedRoute
+  ) {
+  }
+
+  ngOnInit(): void {
+    const fetchModule = this.moduleContentService.fetchLesson(this.route.snapshot.paramMap.get('lessonId')).pipe(
+      tap(data => {
+        this.lesson = data;
+        this.moduleProgress = this.moduleService.getModule();
+        const fetchLesson = this.moduleProgress.lessons.filter(lesson => lesson.lessonId === this.lesson.lessonId);
+        this.lessonDone = fetchLesson[0].complete;
```

Comment:
Essa sintaxe não é a melhor prática. Se não me engano, essa sintaxe funciona melhor:
`const [fetchLesson] =  this.moduleProgress.lessons.filter(lesson => lesson.lessonId === this.lesson.lessonId);`

Assim a constante fetchLesson já vai retornar a Lesson que você quer

Essa sintaxe significa que você espera que retorne um array com só um elemento.

Reply:
Perfeito!!

Reply:
Corrigido!